include/mqtt/endpoint.hpp

```diff
@@ -3128,7 +3129,7 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
      */
     void connack(
         bool session_present,
-        std::uint8_t reason_code,
+        variant<connect_return_code, v5::connect_reason_code> reason_code,
```

Comment:
Note that because the connect_return_code values and the v5::connect_reason_code values don't over lap at all, I've used a variant here to allow the end user to provide either.

When mqtt_cpp::endpoint decides whether to send a v3, or a v5, message, if the provided value isn't the right kind, an exception will be thrown by mqtt::variant.

Reply:
When mqtt_cpp::endpoint calls the v5_connack_handler, it only uses v5::connect_reason_code. When mqtt_cpp::endpoint calls the v3 connack_handler, it only uses connect_return_code.

Reply:
Ok, I've checked the code. I think that it is good.